tutorials/custom-item-models.md

```diff
@@ -0,0 +1,416 @@
+```json
+{
+	"format_version": "1.8.0",
+	"animations": {
+		"animation.<your_item>.first_person.hold": {
+			"loop": true,
+			"bones": {
+				"<root_bone_of_your_item>": {
+					"rotation": [87.5, 52.5, -7.5],
+					"position": [0, 7, 0]
+				}
+			}
+		}
+	}
+}
+```
+Note: it is highly likely that these numbers will **not** work for you. You must play around with various numbers to get the right ones.
+Note: your item geometry must be contained within a root bone that can be easily manipulated.
+In our case, the animation will look like this:
+```json
+{
+	"format_version": "1.8.0",
+	"animations": {
+		"animation.drill.first_person.hold": {
+			"loop": true,
+			"bones": {
+				"drill": {
+					"rotation": [87.5, 52.5, -7.5],
+					"position": [0, 7, 0]
+				}
+			}
+		}
+	}
+}
+```
+Now, let's add the decorative animation for our drill:
+```json
+{
+	"format_version": "1.8.0",
+	"animations": {
+		"animation.drill.hold": {
+			"loop": true,
+			"bones": {
+				"rotator": {
+					"rotation": [0, 0, "query.anim_time * 850"]
+				},
+				"inner_rotor": {
+					"rotation": [0, 0, "query.anim_time * -950"]
+				}
+			}
+		},
+		"animation.drill.first_person.hold": {
+			"loop": true,
+			"bones": {
+				"drill": {
+					"rotation": [87.5, 52.5, -7.5],
+					"position": [0, 7, 0]
+				}
+			}
+		}
+	}
+}
+```
+This animation will spin the drill really quickly to simulate an actual drill.
+Now, let's add these animations to the player client definition file. Add a new entity to the `animations` object of your `player.entity.json` file like this:
+```json
+"animations": {
+    ...,
+    "<your_item>_first_person": "animation.<your_item>.first_person.hold" //This is the first-person animation you definde in the previous step
+}
+```
+Next, let's make this animation loop: add a new entry to your `scripts.animate` array like so:
+```json
+"animate": [
+    ...,
+    {
+        //This is the animation you defined in the previous step
+        "<your_item>_first_person": "variable.<your_item> && variable.is_first_person" //This is the variable you defined in the 'pre_animation' array. This animation will only play if the player is in first person and holding the item (p ∧ q)
+    }
+]
+```
+
+In our case, it will look like this:
+```json
+"animate": [
+    ...,
+    {
+        "drill_first_person": "variable.drill && variable.is_first_person"
+    },
+    {
+        "drill": "variable.drill"
+    }
+]
+```
+Note: we have the decorative animation defined here as well.
+
+## Final Touches
+Great! Our item works but there is a small problem: you can see that our item sprite is rendered with the 3D model in the player's hand.
+
+![](/assets/images/tutorials/custom-item-models/drill-item-visible.png)
+
+To fix this, simply set the opacity of the item texture to 1 in a free image editor like GIMP.
```

Comment:
Won't this make the "drill" in the inventory blank?

Reply:
No because Minecraft is a special kind of beast